core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -159,7 +159,8 @@
 
 public class ExpressionInterpreter
 {
-    private static final Map<InListExpression, Set<?>> inListCache = new WeakHashMap<>();
+    private static final Map<InListExpression, Set<?>> inListLiteralCache = new WeakHashMap<>();
```

Comment:
nit: I'm not sure this cache is even used in practice TBH

Reply:
Yes, that's true. This cache isn't used in `planLargeInQuery` benchmark.

Reply:
Also, this cache only gets used when the InPredicate value isn't an expression which seems like a very rare use case. Please correct me if I'm wrong

Reply:
Seems so and seems to have always been so since 5c8b118256d32bd004266b7058589129c2acf24d.
Maybe we should remove the `io.trino.sql.planner.ExpressionInterpreter#inListCache`? cc @martint

Reply:
> Maybe we should remove the io.trino.sql.planner.ExpressionInterpreter#inListCache?

Yes, I think we should do that